make test-e2e fails on push

It does not seem right that only privileged people should be allowed to run the e2e test.
```
make -C controllers/nginx container
make[2]: Entering directory `/home/porridge/Pulpit/coding/go/src/k8s.io/ingress/controllers/nginx'
docker build -t gcr.io/google_containers/nginx-ingress-controller:0.8.4 rootfs
Sending build context to Docker daemon 27.92 MB
Step 1 : FROM gcr.io/google_containers/nginx-slim:0.11
 ---> 1f831b9bc034
Step 2 : RUN DEBIAN_FRONTEND=noninteractive apt-get update && apt-get install -y   diffutils   ssl-cert   --no-install-recommends   && rm -rf /var/lib/apt/lists/*   && make-ssl-cert generate-default-snakeoil --force-overwrite
 ---> Using cache
 ---> d503fdd37ab3
Step 3 : COPY . /
 ---> e8f230d9d3a0
Removing intermediate container 19fa9207856d
Step 4 : RUN curl -sSL -o /sbin/dumb-init https://github.com/Yelp/dumb-init/releases/download/v1.2.0/dumb-init_1.2.0_amd64 &&   chmod +x /sbin/dumb-init
 ---> Running in ea440bcbb2a1
 ---> e5adf07224f9
Removing intermediate container ea440bcbb2a1
Step 5 : ENTRYPOINT /sbin/dumb-init --
 ---> Running in bdc2f45c6af7
 ---> 6c2c67728394
Removing intermediate container bdc2f45c6af7
Step 6 : CMD /nginx-ingress-controller
 ---> Running in 352d60740191
 ---> 3968d1630603
Removing intermediate container 352d60740191
Successfully built 3968d1630603
make[2]: Leaving directory `/home/porridge/Pulpit/coding/go/src/k8s.io/ingress/controllers/nginx'
make -C controllers/nginx push
make[2]: Entering directory `/home/porridge/Pulpit/coding/go/src/k8s.io/ingress/controllers/nginx'
gcloud docker push gcr.io/google_containers/nginx-ingress-controller:0.8.4
WARNING: The '--' argument must be specified between gcloud specific args on the left and DOCKER_ARGS on the right. IMPORTANT: previously, commands allowed the omission of the --, and unparsed arguments were treated as implementation args. This usage is being deprecated and will be removed in March 2017.
This will be strictly enforced in March 2017. Use 'gcloud beta docker' to see new behavior.
Using 'push gcr.io/google_containers/nginx-ingress-controller:0.8.4' for DOCKER_ARGS.
The push refers to a repository [gcr.io/google_containers/nginx-ingress-controller]
c3a8219e9c1c: Pushing [==================================================>] 48.64 kB
cc537ee68cae: Pushing [==================================================>] 27.89 MB/27.89 MB
5911c4a66511: Pushing [==================================================>] 1.036 MB
6d33e15f21fc: Mounted from google_containers/nginx-slim 
1d2899dd3441: Mounted from google_containers/nginx-slim 
161ae3947447: Waiting 
36e0302572e7: Waiting 
5f70bf18a086: Waiting 
85f69a0fae58: Waiting 
denied: Access denied.
make[2]: *** [push] Error 1
make[2]: Leaving directory `/home/porridge/Pulpit/coding/go/src/k8s.io/ingress/controllers/nginx'
make[1]: *** [docker-push] Error 2
make[1]: Leaving directory `/home/porridge/Pulpit/coding/go/src/k8s.io/ingress'
2016/12/22 09:53:12 e2e.go:278: Step 'build-release' finished in 1m7.73947222s
2016/12/22 09:53:12 e2e.go:137: Something went wrong: error building: error building: error running build-release: exit status 2
exit status 1
make: *** [test-e2e] Error 1
porridge@beczulka:~/Desktop/coding/go/src/k8s.io/ingress$ 
```

--- a/hack/e2e.go
+++ b/hack/e2e.go
@@ -214,7 +214,7 @@ func run(deploy deployer) error {
 func Build() error {
 	// The build-release script needs stdin to ask the user whether
 	// it's OK to download the docker image.
-	cmd := exec.Command("make", "docker-build", "docker-push")
+	cmd := exec.Command("make", "docker-build")
 	cmd.Stdin = os.Stdin
 	if err := finishRunning("build-release", cmd); err != nil {
 		return fmt.Errorf("error building: %v", err)

